### DISCUSSION TITLE
Broken autocompletion for path type command argument

### DISCUSSION THREAD

**Discussion by marcelldls** [fastapi/typer#731] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pathlib import Path

import typer

app = typer.Typer()


@app.command()
def echo_path(name: Path):
    typer.echo(f"Path: {name}")


if __name__ == "__main__":
    app()
```

### Description

Triggering autocompletion on a path command argument returns a " " when using typer, rather than autocompleting from the current directory tree. This works fine in click . My current work around for a Path is to use `autocompletion=lambda: []` in order to get the desired auto-completion behaviour

### Operating System

Linux

### Operating System Details

GNU bash, version 4.4.20(1)-release (x86_64-redhat-linux-gnu)

### Typer Version

0.7.0

### Python Version

3.11.5

### Additional Context

Typer example - https://github.com/marcelldls/typer_path_complete
Click example - https://github.com/marcelldls/click_path_complete

**Comment by YuriiMotov** [comment]
Does anybody still have this issue?
I tested code with current version of Typer (0.17.4) on Ubuntu and it works well.
I'll close this discussion since it's quite old and the issue has likely already been fixed. Please, let me know if this still doesn't work on your system
